client-side/src/components/profileComponents/updateProfileCpmponent.jsx

```diff
@@ -0,0 +1,408 @@
+                <DialogTitle>Edit Profile</DialogTitle>
+                <DialogContent>
+                    <Box sx={{ flexGrow: 1 }} ml={15}>
+                        <Grid container spacing={2}>
+                            <Grid item xs={1} sm={3}>
+                                <Box mt={3.2}>
+                                    <GenericInput
+                                        label="Profile Name"
+                                        name="profileName"
+                                        value={formData.profileName}
+                                        onChange={(e) => handleFieldChange(e)}
+                                        width='100%'
+                                    />
+                                </Box>
+                            </Grid>
+                            <Grid item xs={1} sm={2}>
+                                <Box mt={3.2}>
+                                    <GenericInput
+                                        name="timeStart"
+                                        type="time"
+                                        value={formData.timeProfile.timeStart}
+                                        onChange={(e) => handleFieldChange(e)}
+                                        width='100%'
+                                    />
+                                </Box>
+                            </Grid>
+                            <Grid item xs={1} sm={2}>
+                                <Box mt={3.2}>
+                                    <GenericInput
+                                        name="timeEnd"
+                                        type="time"
+                                        value={formData.timeProfile.timeEnd}
+                                        onChange={(e) => handleFieldChange(e)}
+                                        width='100%'
+                                    />
+                                </Box>
+                            </Grid>
+                            <Grid item xs={1} sm={2}>
+                                <Box mt={3.2}>
+                                    <Select
+                                        name="statusBlockedSites"
+                                        label="Status Blocked Sites"
+                                        value={formData.statusBlockedSites}
+                                        onChange={(e) => handleFieldChange(e)}
+                                        options={[
+                                            { text: 'Black List', value: 'black list' },
+                                            { text: 'White List', value: 'white list' }
+                                        ]}
+                                        widthOfSelect='4.5cm'
+                                    />
+                                </Box>
+                            </Grid>
+                            <Grid item xs={12}>
```

Comment:
edit profile should enable only editing of profile name, start & end times, and status. editing websites should be only through inline edit in the grid